iroh-relay/src/protos/common.rs

```diff
@@ -69,13 +79,15 @@ impl FrameType {
 
     /// Parses the frame type (as a QUIC-encoded varint) from the first couple of bytes given
     /// and returns the frame type and the rest.
-    pub(crate) fn from_bytes(bytes: Bytes) -> Option<(Self, Bytes)> {
-        let mut cursor = std::io::Cursor::new(&bytes);
-        let tag = VarInt::decode(&mut cursor).ok()?;
-        let tag_u32 = u32::try_from(u64::from(tag)).ok()?;
-        let frame_type = FrameType::from(tag_u32);
-        let content = bytes.slice(cursor.position() as usize..);
-        Some((frame_type, content))
+    pub(crate) fn from_bytes(buf: &mut impl Buf) -> Result<Self, FrameTypeError> {
+        let tag = VarInt::decode(buf).ok().context(UnexpectedEndSnafu)?;
+        let tag_u32 = u32::try_from(u64::from(tag))
+            .ok()
+            .context(UnknownFrameTypeSnafu { tag })?;
+        let frame_type = FrameType::try_from(tag_u32)
+            .ok()
+            .context(UnknownFrameTypeSnafu { tag })?;
```

Comment:
Is there any reason you went with u32 for the frame type?  Seeing these shenanigans here seems needlessly complex?  Why didn't you go with a u64?  

Though also, this is purely a code thing.  Changing this changes nothing on the wire.

Reply:
That's what the `num_enum::TryFromPrimitive` macro gives me.
And I think the reason for that is that that's what rust does during compilation usually (it represents enums as u32 at runtime).